php/functions.php

```diff
@@ -47,30 +42,126 @@ function addParagraphs(array $paragraphs) :string {
  * @param $addSubmit string inserts the string in the text area to the database and pairing it with an `id` and a `deleted` number
  */
 function addParagraphToDb (PDO $db, string $addSubmit) : void{
-    $query = $db->prepare("INSERT INTO `about_me` (`paragraph`) VALUES (:newParagraph);");
-    $query->bindParam(':newParagraph', $addSubmit);
-    $query->execute();
+    if (empty(trim($addSubmit))){
+        header('Location: admin_page.php');
+    } else if (strlen($addSubmit) < 1000) {
+        $query = $db->prepare("INSERT INTO `about_me` (`paragraph`) VALUES (:newParagraph);");
+        $query->bindParam(':newParagraph', $addSubmit);
+        $query->execute();
+    } else {
+        header('Location: admin_page.php');
+    }
 }
 
 
+//function addParagraphToDb (PDO $db, string $addSubmit) : void{
+//    if (strlen($addSubmit) < 1000) {
+//        $query = $db->prepare("INSERT INTO `about_me` (`paragraph`) VALUES (:newParagraph);");
+//        $query->bindParam(':newParagraph', $addSubmit);
+//        $query->execute();
+//    } else {
+//        header('Location: admin_page.php');
+//    }
+//}
 
-function editDropDown (PDO $db){
-    $query = $db->prepare("SELECT `id`, `paragraph` FROM `about_me`;");
-    $query->execute();
-    return $query->fetchAll();
-}
-
-function editParagraphDropdown ($editParagraphs){
+/**this function returns an <option> tag with the paragraph value inside the opening tag and the content of that id
+ *
+ * @param array $paragraphs ignore the name, its a placeholder but it is the list of the paragraphs in the database
+ *
+ * @return string returns a list of <options> in  the dropdown
+ */
+function editParagraphDropdown (array $paragraphs) : string {
     $paragraphList = '';
-    foreach ($editParagraphs as $paragraph){
-        $paragraphList .= '<option' . $paragraph['id'] . '>' . $paragraph['paragraph'] . '</option>'''
+    $i = 1 ;
+    foreach ($paragraphs as $paragraph){
+        if (is_string($paragraph['paragraph']) && array_key_exists('paragraph', $paragraph)) {
+            $charPreview = substr($paragraph['paragraph'], 0, 20);
+            $paragraphList .= '<option value=' . $paragraph['id'] . '>paragraph ' . $i++ . ' ' . $charPreview . '</option>';
+        } else {
+            $paragraphList .= '';
+        }
     }
     return $paragraphList;
 }
 
+/**
+ * this function retrieves the paragraph that is chosen in the dropdown list
+ *
+ * @param $db PDO calls the database from the db_query because this function requires it
+ *
+ * @param $editChoice string outputs a number of the id that will be used by the WHERE in MySQL
+ *
+ * @return mixed retrieves one row of the result by the SELECT in MySQL
+ */
+function getEdit(PDO $db, string $editChoice){
+    if ($editChoice == "choose"){
+        header('Location: admin_page.php');
```

Comment:
Don't think you should have a redirect in here!

Reply:
changed it like all the other functions with redirect, the if statement is in admin_page